Firestore/Source/Local/FSTLocalStore.mm

```diff
@@ -165,17 +168,17 @@ - (MaybeDocumentMap)userDidChange:(const User &)user {
   [self startMutationQueue];
 
   return self.persistence.run("NewBatches", [&] {
-    std::vector<FSTMutationBatch *> newBatches = _mutationQueue->AllMutationBatches();
+    std::vector<MutationBatch> newBatches = _mutationQueue->AllMutationBatches();
 
     // Recreate our LocalDocumentsView using the new MutationQueue.
     _localDocuments = absl::make_unique<LocalDocumentsView>(_remoteDocumentCache, _mutationQueue,
                                                             [_persistence indexManager]);
 
     // Union the old/new changed keys.
     DocumentKeySet changedKeys;
-    for (const std::vector<FSTMutationBatch *> &batches : {oldBatches, newBatches}) {
-      for (FSTMutationBatch *batch : batches) {
-        for (const Mutation &mutation : [batch mutations]) {
+    for (const std::vector<MutationBatch> &batches : {oldBatches, newBatches}) {
```

Comment:
Unrelated to this PR: this is nice syntax, but unfortunately it copies the vectors (initializer lists always copy the arguments), perhaps split into two loops (one over each vector)? (Alternatively, you could store pointers to the vectors in the initializer list)

https://wandbox.org/permlink/E0dCP1ByUFhlrrpJ

Reply:
Went with the pointers--that's much easier.